### PR TITLE
[MOUNTMGR] HasDriveLetter(): Fix and simplify

### PR DESCRIPTION
**Pull request by SergeGautherie**
## Purpose

Same fix as bf6215c6019ad14352a9afcc7fc28095bebd2673 ;-)

### CODE REVIEW

**Comment by HBelusca** [comment]
Cc @HeisSpiter

File: drivers/filters/mountmgr/device.c [hunk 1]
```diff
@@ -506,13 +506,13 @@ MountMgrNextDriveLetterWorker(IN PDEVICE_EXTENSION DeviceExtension,
     DeviceInformation->LetterAssigned =
     DriveLetterInfo->DriveLetterWasAssigned = TRUE;
 
-    /* Browse all the symlink to see if there's already a drive letter */
```
Comment (HeisSpiter) [inline drivers/filters/mountmgr/device.c]: "the" is legit here.

File: drivers/filters/mountmgr/mountmgr.c [hunk 2]
```diff
@@ -89,28 +89,19 @@ HasDriveLetter(IN PDEVICE_INFORMATION DeviceInformation)
...
-    /* To have a drive letter, a device must have symbolic links */
-    if (IsListEmpty(&(DeviceInformation->SymbolicLinksListHead)))
-    {
-        return FALSE;
-    }
-
-    /* Browse all the links untill a drive letter is found */
-    NextEntry = &(DeviceInformation->SymbolicLinksListHead);
-    do
+    /* Browse all symlinks to check if there is at least a drive letter */
```
Comment (HeisSpiter) [inline drivers/filters/mountmgr/mountmgr.c]: "the" is legit here too

File: drivers/filters/mountmgr/mountmgr.c [hunk 3]
```diff
@@ -89,28 +89,19 @@ HasDriveLetter(IN PDEVICE_INFORMATION DeviceInformation)
...
+    NextEntry = DeviceInformation->SymbolicLinksListHead.Flink;
+    while (NextEntry != &(DeviceInformation->SymbolicLinksListHead))
```
Comment (HeisSpiter) [inline drivers/filters/mountmgr/mountmgr.c]: While you're at it, use a for-loop

File: drivers/filters/mountmgr/mountmgr.c [hunk 4]
```diff
@@ -89,28 +89,19 @@ HasDriveLetter(IN PDEVICE_INFORMATION DeviceInformation)
...
     {
         SymlinkInfo = CONTAINING_RECORD(NextEntry, SYMLINK_INFORMATION, SymbolicLinksListEntry);
 
-        if (SymlinkInfo->Online)
+        if (IsDriveLetter(&(SymlinkInfo->Name)) && SymlinkInfo->Online)
```
Comment (HeisSpiter) [inline drivers/filters/mountmgr/mountmgr.c]: And remove extra parenthesis.

File: drivers/filters/mountmgr/mountmgr.c (hunk 2 above)
Comment (SergeGautherie) [reply drivers/filters/mountmgr/mountmgr.c]: Re-added. (Suggest/Fix the whole sentence, if not what you expect.)

File: drivers/filters/mountmgr/mountmgr.c (hunk 3 above)
Comment (SergeGautherie) [reply drivers/filters/mountmgr/mountmgr.c]: Done.

File: drivers/filters/mountmgr/mountmgr.c (hunk 4 above)
Comment (SergeGautherie) [reply drivers/filters/mountmgr/mountmgr.c]: Done.

File: drivers/filters/mountmgr/device.c (hunk 1 above)
Comment (SergeGautherie) [reply drivers/filters/mountmgr/device.c]: Re-added. (Suggest/Fix the whole sentence, if not what you expect.)

Comment (learn-more) [reply drivers/filters/mountmgr/device.c]: tbh I don't see why this comment needs changing.
Is this just changing stuff to get the linecount up?

**Comment by learn-more** [comment]
'Same fix' yet the change is way bigger....

**Pull request merged**